wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/JavaxExceptionMapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server;
+
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+import javax.inject.Singleton;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.NotAcceptableException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import org.eclipse.che.api.core.rest.shared.dto.ServiceError;
+import org.eclipse.che.dto.server.DtoFactory;
+
+@Provider
+@Singleton
+public class JavaxExceptionMapper implements ExceptionMapper<WebApplicationException> {
```

Comment:
Since we can only throw instances of `WebApplicationEcxeption` from EntityProviders, they're mapped into 500-s by default. So with writing own mapper so we can use `javax.ws.rs` exceptions as an equivavlent to API ones.

If that approach is approved, this class should contain all package exceptions  and probably can be moved to `che-core-rest`